pkg/filters/args.go

```diff
@@ -55,10 +58,21 @@ func (filter *ArgFilter) Filter(eventID events.ID, args []trace.Argument) bool {
 		if !found {
 			return false
 		}
-		// TODO: use type assertion instead of string conversion
-		if argName != "syscall" {
-			argVal = fmt.Sprint(argVal)
+
+		argVal = fmt.Sprint(argVal)
+		if argName == "syscall" {
```

Comment:
argVal is coming from Filter() args parameter, in other words, from the event.

Reply:
I am not talking about `argVal`.
I am talking about what we compare it to.
I think we should transform the filtered syscall name to a number, instead of transforming the received syscall number to its name.

Reply:
@AlonZivony you've opened my eyes. Thanks. Going to push the suggested solution.